### DISCUSSION TITLE
[New Environment variable] add an Environment variable like VCPKG_DISABLE to disable vcpkg integration once but not global

### DISCUSSION THREAD

**Discussion by congzhangzh** [microsoft/vcpkg#37546] [Ideas]
### Is your feature request related to a problem? Please describe.

node windows compile failed if zlib is installed by vcpkg and vcpkg integration install is done

I can confirm that because the node link error gone after **vcpkg integration remove**

Ref: https://github.com/nodejs/node/issues/24448

### Proposed solution

add  VCPKG_DISABLE to disable vcpkg integration for the child process, because my other c++ project depend on vcpkg integration.

this can be done either skip on user side, or system side:
![image](https://github.com/microsoft/vcpkg/assets/13283869/e69a7db1-ad63-4ec4-b554-95e56c10d1ce)
![image](https://github.com/microsoft/vcpkg/assets/13283869/2b1d09ba-b135-4892-a8b8-3b0409a2af31)

### Describe alternatives you've considered

do vcpkg integration remove and run other c++ project with cmake toolchain

### Additional context

https://learn.microsoft.com/en-us/vcpkg/users/config-environment

**Comment by FrankXie05** [comment]
@congzhangzh Thanks for posting this issue.
For each project, their property interface has a corresponding button `Use Vcpkg` to enable vcpkg.
If you only want some sub-projects to use vcpkg, you only need to open the corresponding property interface and set `Use Vcpkg` to `No`.

![image](https://github.com/microsoft/vcpkg/assets/65999885/913f724a-401b-4484-9251-fca21d50a1d8)

**Comment by congzhangzh** [comment]
@FrankXie05 tks for your replay, that's a way, but that's not always possible, especially for batch shell script.

for example, if I can pass something to msbuild I can do 
```bat
msbuild ... /p:UseVcpkg=false
```
I already do this: https://github.com/congzhangzh/node/blob/66a6e5ee463491b6c0416d4393cd27bf985747a5/vcbuild.bat#L335

but if this happened for some inner call, for example, A shell call B, B shell call C, and C need to disable vcpkg, and A=>B can not accept and pass something to C's msbuild, like B&C is some third party application or shell, I can do nothing.

**Comment by congzhangzh** [comment]
something like Directory.build.props is another way, but that's another story:)

**Comment by FrankXie05** [comment]
@congzhangzh I'll turn it into a discussion and let's communicate here. :)

**Comment by FrankXie05** [comment]
>A shell call B, B shell call C, and C need to disable vcpkg, and A=>B can not accept and pass something to C's msbuild, like B&C is some third party application or shell, I can do nothing.

From my understanding: the core lies in whether to inherit the properties used by vcpkg.

Under normal circumstances, we can manually modify or pass the value of `Use vcpkg`.

For the B&C project in the example, if we cannot make any modifications and transfers to its properties, use a command similar to `Integrate install --B or C` to transfer the `.props` and `.targets` files to make initial modifications.

@congzhangzh
